_includes/master/non-helm-manifests/flannel-migration/migration-job.yaml

```diff
@@ -0,0 +1,167 @@
+# Include a clusterrole for the kube-controllers component,
+# and bind it to the flannel-migration-controller serviceaccount.
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: flannel-migration-controller
+rules:
+  # Nodes are watched to monitor for deletions.
+  - apiGroups: [""]
+    resources:
+      - nodes
+    verbs:
+      - watch
+      - list
+      - get
+      - patch
+      - update
+  # Nodes are watched to monitor for deletions.
+  - apiGroups: [""]
+    resources:
+      - nodes/status
+    verbs:
+      - get
+      - update
+  # Pods are created/deleted.
+  - apiGroups: [""]
+    resources:
+      - pods
+    verbs:
+      - get
+      - list
+      - create
+      - delete
+  # Configmaps are updated.
+  - apiGroups: [""]
+    resources:
+      - configmaps
+    verbs:
+      - get
+      - update
+  - apiGroups: [""]
+    resources:
+      - pods/eviction
+    verbs:
+      - create
+  # Daemonset are watched to monitor for deletions.
+  - apiGroups: ["apps", "extensions"]
+    resources:
+      - daemonsets
+    verbs:
+      - get
+      - delete
+      - update
+  # IPAM resources are manipulated when nodes are deleted.
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - ippools
+    verbs:
+      - get
+      - list
+      - create
+      - update
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - ipamconfigs
+      - blockaffinities
+      - ipamblocks
+      - ipamhandles
+    verbs:
+      - get
+      - list
+      - create
+      - update
+      - delete
+  # Needs access to update clusterinformations.
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - clusterinformations
+    verbs:
+      - get
+      - create
+      - update
+  # Needs access to update felixconfigurations.
+  - apiGroups: ["crd.projectcalico.org"]
+    resources:
+      - felixconfigurations
+    verbs:
+      - get
+      - create
+      - update
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: flannel-migration-controller
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: flannel-migration-controller
+subjects:
+- kind: ServiceAccount
+  name: flannel-migration-controller
+  namespace: kube-system
+
+---
+# See https://github.com/projectcalico/kube-controllers
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: flannel-migration
+  namespace: kube-system
+  labels:
+    k8s-app: flannel-migration-controller
+spec:
+  backoffLimit: 10
```

Comment:
I put down number of retries to 10. Not sure what is the best. WDYT? @caseydavenport @lwr20